File: src/core/validators/CommonSimpleType/CommonShortMinValueMustNotBeGreaterThanMaxValue.cs
using System;
using MetaEd.Grammar.Antlr;

namespace MetaEd.Core.Validator.CommonSimpleType
{
    public class CommonShortMinValueMustNotBeGreaterThanMaxValue : ValidationRuleBase<MetaEdGrammar.CommonShortContext>
    {
        public override bool IsValid(MetaEdGrammar.CommonShortContext context)
        {
            if (context.minValue() == null || context.maxValue() == null) return true;
            
            // if there are convert exceptions, let it bomb out -- language parser should have handled
            var minValue = Convert.ToInt32(context.minValue().MinValue());
            var maxValue = Convert.ToInt32(context.maxValue().MaxValue());

            return minValue <= maxValue;
        }

        public override string GetFailureMessage(MetaEdGrammar.CommonShortContext context)
        {
            return string.Format("Common Short '{0}' has min value greater than max value.", context.commonShortName().GetText());
        }
    }
}

File: src/core/validators/CommonSimpleType/CommonShortMinValueMustNotBeGreaterThanMaxValue.ts
import { ValidationRuleBase } from "../ValidationRuleBase";
export class CommonShortMinValueMustNotBeGreaterThanMaxValue extends ValidationRuleBase<MetaEdGrammar.CommonShortContext>
{
    public isValid(context: MetaEdGrammar.CommonShortContext): boolean {
        if (context.minValue() == null || context.maxValue() == null)
            return true;
        var minValue = Number(context.minValue().MinValue());
        var maxValue = Number(context.maxValue().MaxValue());
        return minValue <= maxValue;
    }
    public getFailureMessage(context: MetaEdGrammar.CommonShortContext): string {
        return `Common Short '${context.commonShortName().GetText()}' has min value greater than max value.`;
    }
}
